src/methods/modular_programs/easy/q4/MainTest.java

```diff
@@ -0,0 +1,62 @@
+package methods.modular_programs.easy.q4;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.utils.MethodUtil;
+import global.variables.Clause;
+import global.variables.clauses.DoubleLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Parsons
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The values after the decimal point for the first variable are "),
+                new DoubleLiteral("first"),
+                new NewLine(),
+                new StringLiteral("The values after the decimal point for the second variable are "),
+                new DoubleLiteral("second"),
+                new NewLine(),
+                new StringLiteral("The values after the decimal point for the third variable are "),
+                new DoubleLiteral("third")
+        };
+    }
+
+    public void runMain() {
+        ReturnChange.main(new String[0]);
+    }
+
+    static Stream<Arguments> valueAfterCalcInputProvider() {
+        return Stream.of(
+                Arguments.of(3.1, 0.1),
+                Arguments.of(67.2175396, 0.2175396),
+                Arguments.of(0.34, 0.34),
+                Arguments.of(0.0, 0.0),
+                Arguments.of(-12.523, -0.523),
+                Arguments.of(0.1, 0.1),
+                Arguments.of(-0.1, -0.1));
+    }
+
+    @ParameterizedTest
+    @MethodSource("valueAfterCalcInputProvider")
+    void correctValueAfterCalcMethod(double in, double decimals) throws Throwable {
+        Object output = MethodUtil.invokeIfMethodExists(ReturnChange.class, "valueAfterCalc", new Object[]{in}, double.class);
+        CustomAssertions._assertEquals(decimals, output, 0.00001, "Your valueAfterCalc method does not correctly extract the numbers after the decimal.");
```

Comment:
I think that the delta should be even smaller because there really should be no discrepancy in the number here, as they are just extracting the decimal value